frontend/src/citizen-frontend/calendar/CalendarListView.tsx

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React from 'react'
+import _ from 'lodash'
+import { FixedSpaceColumn } from 'lib-components/layout/flex-helpers'
+import { DailyReservationData } from './api'
+import WeekElem, { WeekProps } from './WeekElem'
+
+export interface Props {
+  data: DailyReservationData[]
+}
+
+export default React.memo(function CalendarListView({ data }: Props) {
+  const weeklyData = data.reduce((weekly, daily) => {
+    const last = _.last(weekly)
+    if (last === undefined || daily.date.getIsoWeek() !== last.weekNumber) {
+      return [...weekly, { weekNumber: daily.date.getIsoWeek(), data: [daily] }]
+    } else {
+      return [
+        ..._.dropRight(weekly),
+        {
+          ...last,
+          data: [...last.data, daily]
+        }
+      ]
+    }
+  }, [] as WeekProps[])
```

Comment:
I would extract this whole reduce into a function with a descriptive name like `partitionByWeek` because it's kind of hard to understand what's going on.